src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/CVTextCriteriaEvaluator.cs

```diff
@@ -0,0 +1,298 @@
+                var resultsString = cvTextResults == null ? "null":$"\n[{string.Join(",\n", cvTextResults)}]";
+                if (cvTextResults != null)
+                {
+                    RGDebug.LogDebug($"CVTextCriteriaEvaluator - Matched - botSegment: {segmentNumber} - cvTextResults: {resultsString}");
+                }
+            }
+
+            if (priorResults == null && cvTextResults != null)
+            {
+                // Evaluate the results if we have some
+
+                // Detailed evaluation of the results vs our criteria, this is currently n^3 (loop in a loop in a loop), but needs to be due to allowing partial word matching
+                // we do our best to make this faster by removing from the lists as we match to reduce future iterations
+                for (var i = criteriaList.Count - 1; i >= 0; i--)
+                {
+                    var criteria = criteriaList[i];
+                    if (!criteria.transient || !criteria.Replay_TransientMatched)
+                    {
+                        var criteriaData = criteria.data as CVTextKeyFrameCriteriaData;
+
+                        // setup a tracker of each text part and whether it is found.. remove from this as we match to improve performance and as a way of marking what was found
+                        // We treat a criteria multi word phrase as separate words that must ALL exist within the defined rect
+                        // If no rect is defined, all the words must simply exist on the screen, but their relative positions are then not considered
+                        var textParts = criteriaData.text.Trim().Split(' ').Select(a => a.Trim()).Where(a => a.Length > 0).ToList();
+
+                        var cvTextResultsCount = cvTextResults.Count;
+                        for (var k = 0; k < cvTextResultsCount && textParts.Count > 0; k++)
+                        {
+                            var cvTextResult = cvTextResults[k];
+                            var cvText = cvTextResult.text.Trim();
+
+                            if (criteriaData.textCaseRule == TextCaseRule.Ignore)
+                            {
+                                cvText = cvText.ToLower();
+                            }
+
+                            for (var j = textParts.Count - 1; j >= 0; j--)
+                            {
+                                var matched = false;
+                                var textToMatch = textParts[j];
+
+                                if (criteriaData.textCaseRule == TextCaseRule.Ignore)
+                                {
+                                    textToMatch = textToMatch.ToLower();
+                                }
+
+                                switch (criteriaData.textMatchingRule)
+                                {
+                                    case TextMatchingRule.Matches:
+                                        if (textToMatch.Equals(cvText))
```

Comment:
Currently, are there any ordering constraints? For instance, if I search for "Start Game", I will match "Game Start" as well, correct? That's fine for now I think but I just wanted to confirm.

Reply:
since the words are found within the bounding area right now, correct.. their relative location/position within that area vs ordering in the input are NOT considered

this decision was made for reasons already discussed in slack thread about 1) simplicity of implementation , 2) naturalness of using it when i searched for a button with multi line text

Reply:
Makes sense!